src/extensions/extension-discovery/is-compatible-extension/is-compatible-extension.ts

```diff
@@ -9,44 +9,14 @@ interface Dependencies {
   appSemVer: SemVer;
 }
 
-export const isCompatibleExtension = ({
-  appSemVer,
-}: Dependencies): ((manifest: LensExtensionManifest) => boolean) => {
-  const { major, minor, patch, prerelease: oldPrelease } = appSemVer;
-  let prerelease = "";
-
-  if (oldPrelease.length > 0) {
-    const [first] = oldPrelease;
-
-    if (first === "alpha" || first === "beta" || first === "rc") {
-      /**
-       * Strip the build IDs and "latest" prerelease tag as that is not really
-       * a part of API version
-       */
-      prerelease = `-${oldPrelease.slice(0, 2).join(".")}`;
-    }
-  }
-
-  /**
-   * We unfortunately have to format as string because the constructor only
-   * takes an instance or a string.
-   */
-  const strippedVersion = new SemVer(
-    `${major}.${minor}.${patch}${prerelease}`,
-    { includePrerelease: true },
-  );
+export const isCompatibleExtension = (deps: Dependencies): ((manifest: LensExtensionManifest) => boolean) => {
+  const { major: appMajor, minor: appMinor } = deps.appSemVer;
 
   return (manifest: LensExtensionManifest): boolean => {
-    if (manifest.engines?.lens) {
-      /**
-       * include Lens's prerelease tag in the matching so the extension's
-       * compatibility is not limited by it
-       */
-      return semver.satisfies(strippedVersion, manifest.engines.lens, {
-        includePrerelease: true,
-      });
-    }
+    const { major: extMajor, minor: extMinor } = new SemVer(manifest.engines.lens, {
+      loose: true,
+    });
 
-    return false;
+    return semver.intersects(`>=${appMajor}.${appMinor}`, `<${extMajor}.${extMinor + 1}`);
```

Comment:
So if the extension has the value of `lens: "5.2"` and the current version is `5.6.1` this would return `false`, right?

Reply:
_Yes:_
<img width="595" alt="Screenshot 2022-05-05 at 18 44 41" src="https://user-images.githubusercontent.com/6377066/166961687-4f79ca7f-6ee4-43b5-a223-b450308020c3.png">

But.. using raw version as `5.2` throws error and some version normalization needed before creating instance.
<img width="592" alt="image" src="https://user-images.githubusercontent.com/6377066/166962045-b7c2072f-098c-4afe-b344-809e0bdf502e.png">

Reply:
But we want `5.2` and `^5.2` to both be valid ways of specifying the range (and to mean the same thing). We want it to return `true` in that case.

Furthermore, we currently accept `>=5.2` and other such semver ranges. So we need to convert them as well.

Reply:
_Test in chrome's devtools:_

```ts
// Test code for devtools console in open lens
var {major: appMajor, minor: appMinor} = new semver.SemVer("5.6.1");
var {major: extMajor, minor: extMinor} = new semver.SemVer("5.2.0");
semver.intersects(`>=${appMajor}.${appMinor}`, `<${extMajor}.${extMinor + 1}`)
```

```ts
// bootstrap.tsx -- add to the end of fle
import semver from "semver";
export {semver};
```